libs/cart/src/injection-tokens/cart-item-state-debounce-time.ts

```diff
@@ -0,0 +1,9 @@
+import { InjectionToken } from '@angular/core';
+
+/**
+ * An injection token for setting how long (in ms) a cart item remains in a temporary state before
+ * reverting back to a default state. These temporary states are set after mutating or adding a cart item.
+ * Daffodil has a timer that resets the state of all cart items after this debounce time, but the timer will
+ * reset if a new item addition or mutation occurs. The default is 4000.
```

Comment:
Its unclear whether a new addition or mutation will cause all the timers to reset or only the timer that pertains to the affected cart item. I would think the timer would decay separately for each cart item in state but that doesn't seem to be how it works.

Reply:
Yea, I explain in the PR description why I decided to go with a global reset instead of tracking the state changes individually. I'll take a look at the doc here and try to make it more clear though

Reply:
oh sorry I didn't see that. I don't think the implementation would be that complex, but maybe I'm not thinking far enough ahead. Couldn't you just pass the `item_id` to the reset action? I think the reducer part of that would be straightfoward.

The only uncertainty I have is how to fork the stream such that each cart item add/update can be debounced separately.

Reply:
Yea, I think the tough part is managing individual debounced streams. Imagine the scenario were cartItem1 is added, then cartItem2 is added quickly after (before the debounce timer is complete). I think this should certainly be possible with like some sort of queue or something. But then add in a third action of updating the quantity of cartItem1 directly after adding cartItem2, and we can no longer think of each cart item add/update as a completely separate debounce timer. In this scenario, if we treat each action as a separate debounce timer, the state of cartItem1 would go:
cartItem1 add -> cart item 1 state becomes "New" -> start cartItemAdd1 debounce timer
cartItem2 add -> cart item 2 state becomes "New" -> start cartItemAdd2 debounce timer
cartItem1 update qty -> cart item 1 state becomes "Mutating" then becomes "Updated" -> start cartItemUpdate1 debounce timer
cartItemAdd1 debounce timer ends -> cart item 1 state becomes "Default" prematurely when it should be "Updated" for the remaining cartItemUpdate1 debounce timer.

So instead, we would need to give each cart item a debounce timer that would somehow reset when a new action was performed on that cart item. I'm not saying this is impossible, but I compared this complexity to the downside of combining all cart item interactions into a single debounce timer and thought it was worth the trade off. That trade off is just that cartItem2 in the example above will remain in the "New" state for longer than 4 seconds. It will remain in that state until the cartItem1 update state (from the third action) finishes its 4 second debounce timer. And that doesn't seem that bad too me.